### PR TITLE
feat: expose additional options for registering views

### PR DESCRIPTION
**Pull request by CompuIves**
I'm preparing us for phase 2 of the implementation of VSCode: replacing our sidebar with the one from VSCode.

To enable this, we'll need to render some of our React components inside the sidebar, sometimes as a new view item (separate view in activity bar). With this PR, I'm extending the capabilities of `registerCustomView` to allow for:

1. More finegrained options (in terms of order, collapsed, etc...)
2. Ability to register a new item in existing views (like the explorer), instead of creating a new view

Let me know if this change makes sense to you!

### CODE REVIEW

File: src/service-override/views.ts [hunk 1]
```diff
@@ -388,12 +388,16 @@ interface CustomViewOption {
...
+  canToggleVisibility?: boolean
+  hideByDefault?: boolean
+  collapsed?: boolean
 }
 
 function registerCustomView (options: CustomViewOption): IDisposable {
   const iconUrl = options.icon != null ? URI.parse(options.icon) : undefined
 
-  const VIEW_CONTAINER = Registry.as<IViewContainersRegistry>(ViewExtensions.ViewContainersRegistry).registerViewContainer({
+  const VIEW_CONTAINER = options.viewContainer ?? Registry.as<IViewContainersRegistry>(ViewExtensions.ViewContainersRegistry).registerViewContainer({
```
Comment (CGNonofr) [inline src/service-override/views.ts:400]: How did you create the view container?

Comment (CGNonofr) [reply src/service-override/views.ts:400]: btw, you may want to NOT deregistering the view container then line 456

Comment (CompuIves) [reply src/service-override/views.ts:400]: Good one. I get them through `IViewDescriptorService`. I run this:

```
viewDescriptorService
            .getViewContainersByLocation(0)
```

Where 0 corresponds to the sidebar.

**Comment by CompuIves** [comment]
Updated the disposal!

File: src/service-override/views.ts (hunk 1 above)
Comment (CGNonofr) [reply src/service-override/views.ts:400]: Ok I have no idea why the variable is in upper case, do you mind changing it?

Comment (CompuIves) [reply src/service-override/views.ts]: Sounds good, it's update!

**Review by CGNonofr — APPROVED** [review]

**Pull request merged**